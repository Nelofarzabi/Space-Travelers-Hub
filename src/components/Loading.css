.loading--container {
  margin: 100px 0;
  display: flex;
  flex-direction: column;
  gap: 0.7rem;
  align-items: center;
  justify-content: center;
}

.Loading--header {
  font-size: 2em;
  text-transform: capitalize;
}
.loading--container p {
  font-size: 1.3rem;
  letter-spacing: 2px;
}