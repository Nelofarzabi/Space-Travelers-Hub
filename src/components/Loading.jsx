import './Loading.css';
import PropTypes from 'prop-types';

const Loading = ({ type }) => (
  <div className="loading--container">
    <h4 className="Loading--header">{`Loading ${type} ...`}</h4>
    <p>Please hold on...</p>
  </div>
);

Loading.propTypes = {
  type: PropTypes.string.isRequired,
};
export default Loading;
