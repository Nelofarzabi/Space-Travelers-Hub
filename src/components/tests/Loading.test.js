import { render } from '@testing-library/react';
import Loading from '../Loading';

describe('Test Error Component', () => {
  test('Loading component renders correctly', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
