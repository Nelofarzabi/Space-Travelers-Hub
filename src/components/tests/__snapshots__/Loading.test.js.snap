// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Error Component Loading component renders correctly 1`] = `
<div>
  <div
    class="loading--container"
  >
    <h4
      class="Loading--header"
    >
      Loading undefined ...
    </h4>
    <p>
      Please hold on...
    </p>
  </div>
</div>
`;
